ce/ce/artifacts/activation.ts

```diff
@@ -1,67 +1,561 @@
-/* eslint-disable keyword-spacing */
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { delimiter } from 'path';
+/* eslint-disable prefer-const */
+
+import { lstat } from 'fs/promises';
+import { delimiter, extname, resolve } from 'path';
+import { isScalar } from 'yaml';
+import { undo as undoVariableName } from '../constants';
+import { i } from '../i18n';
+import { Exports } from '../interfaces/metadata/exports';
 import { Session } from '../session';
-import { linq } from '../util/linq';
+import { isIterable } from '../util/checks';
+import { linq, Record } from '../util/linq';
+import { Queue } from '../util/promise';
 import { Uri } from '../util/uri';
 import { toXml } from '../util/xml';
 import { Artifact } from './artifact';
 
+function findCaseInsensitiveOnWindows<V>(map: Map<string, V>, key: string): V | undefined {
```

Comment:
Ouch this really needs our almost_canonical. No change requested.

https://github.com/microsoft/vcpkg-tool/blob/3da30bc77bda131562fd032ce5c4334c9f398cef/src/vcpkg/base/files.cpp#L3087-L3100